Tools/autotest/arducopter.py

```diff
@@ -1671,33 +1671,121 @@ def fly_optical_flow_limits(self):
 
     # fly_autotune - autotune the virtual vehicle
     def fly_autotune(self):
+        """Test autotune mode"""
 
-        self.takeoff(10)
+        ex = None
+        self.context_push()
+        try:
+            rlld = self.get_parameter("ATC_RAT_RLL_D")
+            rlli = self.get_parameter("ATC_RAT_RLL_I")
+            rllp = self.get_parameter("ATC_RAT_RLL_P")
+            self.takeoff(10)
 
-        # hold position in loiter
-        self.mavproxy.send('mode autotune\n')
-        self.wait_mode('AUTOTUNE')
+            # hold position in loiter
+            self.mavproxy.send('mode autotune\n')
+            self.wait_mode('AUTOTUNE')
 
-        tstart = self.get_sim_time()
-        sim_time_expected = 5000
-        deadline = tstart + sim_time_expected
-        while self.get_sim_time_cached() < deadline:
-            now = self.get_sim_time_cached()
-            m = self.mav.recv_match(type='STATUSTEXT',
-                                    blocking=True,
-                                    timeout=1)
-            if m is None:
-                continue
-            self.progress("STATUSTEXT (%u<%u): %s" % (now, deadline, m.text))
-            if "AutoTune: Success" in m.text:
-                self.progress("AUTOTUNE OK (%u seconds)" % (now - tstart))
-                # near enough for now:
-                self.change_mode('LAND')
-                self.wait_disarmed()
-                return
+            tstart = self.get_sim_time()
+            sim_time_expected = 5000
+            deadline = tstart + sim_time_expected
+            while self.get_sim_time_cached() < deadline:
+                now = self.get_sim_time_cached()
+                m = self.mav.recv_match(type='STATUSTEXT',
+                                        blocking=True,
+                                        timeout=1)
+                if m is None:
+                    continue
+                self.progress("STATUSTEXT (%u<%u): %s" % (now, deadline, m.text))
+                if "AutoTune: Success" in m.text:
+                    self.progress("AUTOTUNE OK (%u seconds)" % (now - tstart))
+                    # near enough for now:
+                    self.change_mode('LAND')
+                    self.wait_disarmed()
+                    # check the original gains have been re-instated
+                    if rlld != self.get_parameter("ATC_RAT_RLL_D") or rlli != self.get_parameter("ATC_RAT_RLL_I") or rllp != self.get_parameter("ATC_RAT_RLL_P"):
```

Comment:
Adding prens here will let this not be too long a line.

Reply:
This comment applies on several others in the PR.

Reply:
Fixed